### PR TITLE
Feat(web): Introduce UNSTABLE Toggle component #DS-1345

### PR DESCRIPTION
**Pull request by crishpeen**

### CODE REVIEW

**Review by dlouhak — COMMENTED** [review]
🎩 , I have only a few comments.

During the code review, I came up with a few insights, mostly related to the stable version of this component:

1. In the Figma it is possible to set if the component has icon or not. 
image

2. Should it be possible **change state by clicking on the validation/helper text**? (related to the Checkbox component, where you probably inspired yourself). And I've also noticed that in the example of validation states the danger variant with multiple error messages doesn't have cursor over label.

3. Nitpicking idea for form components: If there would be a composition of various form components, I can imagine that designer would like to have **possibility to set order of the component** (toggle-label or label-toggle), because eg. `Checkbox` has input first and then label. The same applies for transitions between checked/unchecked states.
image

File: packages/web/src/scss/components/UNSTABLE_Toggle/index.html [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+</section>
+
+<section class="UNSTABLE_Section">
+  <h2 class="docs-Heading">Validation State with Validation Text</h2>
+
+  <div class="docs-Stack docs-Stack--stretch">
+
+    <label for="toggle-success" class="UNSTABLE_Toggle UNSTABLE_Toggle--success">
+      <span class="UNSTABLE_Toggle__text">
+        <span class="UNSTABLE_Toggle__label">Toggle Label</span>
```
Comment (curdaj) [inline packages/web/src/scss/components/UNSTABLE_Toggle/index.html:131]: Why is the validation message not used here as well?

**Comment by crishpeen** [comment]
> 🎩 , I have only a few comments.
> 
> During the code review, I came up with a few insights, mostly related to the stable version of this component:
> 
> 1. In the Figma it is possible to set if the component has icon or not. This shouldn't be implemented?
> 
> image
> 2. Should it be possible **change state by clicking on the validation/helper text**? (related to the Checkbox component, where you probably inspired yourself). And I've also noticed that in the example of validation states the danger variant with multiple error messages doesn't have cursor over label.
> 3. Nitpicking idea for form components: If there would be a composition of various form components, I can imagine that designer would like to have **possibility to set order of the component** (toggle-label or label-toggle), because eg. `Checkbox` has input first and then label. The same applies for transitions between checked/unchecked states.
>    image

1. yes, I missed that
2. it acts the same as in checkbox, the danger state with multiple validation texts is a problem and I will create a topic for our thurday talks, because currently we are causing invalid HTML and I don't know what is the best solution. The current state is as elsewhere (not ideal, but working)
3. We decided to do this layout now. In relatively short term we plan to split all form components to small pieces and user will be able to combine them as they want.

File: packages/web/src/scss/components/UNSTABLE_Toggle/_theme.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
...mark-height}" viewBox="0 0 52 44" fill="none"><path fill="#{tokens.$background-basic}" d="M48.675 14.33a.58.58 0 0 0-.822 0L45 17.178l-2.852-2.852a.58.58 0 1 0-.823.823L44.177 18l-2.852 2.852a.58.58 0 1 0 .822.823L45 18.823l2.852 2.852a.58.58 0 1 0 .823-.823L45.823 18l2.852-2.852a.585.585 0 0 0 0-.817ZM5.25 20.651 3.208 18.61a.574.574 0 0 0-.816 0 .574.574 0 0 0 0 .817l2.444 2.444a.58.58 0 0 0 .822 0l6.184-6.177a.574.574 0 0 0 0-.817.574.574 0 0 0-.817 0L5.25 20.651Z"/><g xmlns="http://www.w3.org/2000/svg" filter="url(#a)"><circle xmlns="http://www.w3.org/2000/svg" cx="26" cy="18" r="10" fill="#{tokens.$background-basic}"/></g><defs><filter id="a" width="44" height="44" x="4" y="0" color-interpolation-filters="sRGB" filterUnits="userSpaceOnUse"><feFlood flood-opacity="0" result="BackgroundImageFix"/><feColorMatrix in="SourceAlpha" result="hardAlpha" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"/><feOffset dy="4"/><feGaussianBlur stdDeviation="6"/><feColorMatrix values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.2 0"/><feBlend in2="BackgroundImageFix" result="effect1_dropShadow_1_81"/><feBlend in="SourceGraphic" in2="effect1_dropShadow_1_81" result="shape"/></filter></defs></svg>');
```
[the remark this answers is not in the document]
Comment (crishpeen) [reply packages/web/src/scss/components/UNSTABLE_Toggle/_theme.scss:14]: I don't use it as an icon. I just exported the decoration from Figma directly.

[the remark this answers is not in the document]
Comment (crishpeen) [reply packages/web/src/scss/components/UNSTABLE_Toggle/_theme.scss:14]: yes

File: packages/web/src/scss/components/UNSTABLE_Toggle/_UNSTABLE_Toggle.scss [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    display: flex;
+    align-items: start;
+    justify-content: space-between;
+    max-width: theme.$max-width;
+    padding-block: form-fields-theme.$gap;
+    margin-block: form-fields-theme.$gap;
+    cursor: cursors.$form-fields;
+}
+
+.UNSTABLE_Toggle--fluid {
```
[the remark this answers is not in the document]
Comment (crishpeen) [reply packages/web/src/scss/components/UNSTABLE_Toggle/_UNSTABLE_Toggle.scss:19]: It is not demonstrated in Figma, but it is one of demands of the issue.

File: packages/web/src/scss/components/UNSTABLE_Toggle/index.html (hunk 1 above)
Comment (crishpeen) [reply packages/web/src/scss/components/UNSTABLE_Toggle/index.html:131]: This an example without validation text. It is the same as in other Form elements

File: packages/web/src/scss/components/UNSTABLE_Toggle/_UNSTABLE_Toggle.scss [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+@use '../../tools/svg';
+
+$_field-name: 'UNSTABLE_Toggle';
+
+.UNSTABLE_Toggle {
+    display: flex;
+    align-items: start;
+    justify-content: space-between;
+    max-width: theme.$max-width;
+    padding-block: form-fields-theme.$gap;
```
[the remark this answers is not in the document]
Comment (crishpeen) [reply packages/web/src/scss/components/UNSTABLE_Toggle/_UNSTABLE_Toggle.scss:14]: imho yes, vertical margin can consume itself

File: packages/web/src/scss/components/UNSTABLE_Toggle/README.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+## Helper Text
+
+```html
+<label for="toggle-helper-text" class="UNSTABLE_Toggle">
+  <span class="UNSTABLE_Toggle__text">
+    <span class="UNSTABLE_Toggle__label">Toggle Label</span>
+    <span class="UNSTABLE_Toggle__helperText">Helper text</span>
+  </span>
+  <input type="checkbox" id="toggle-helper-text" class="UNSTABLE_Toggle__input" name="helper-text" />
```
[the remark this answers is not in the document]
Comment (dlouhak) [reply packages/web/src/scss/components/UNSTABLE_Toggle/README.md:85]: Won't be added anywhere in this README?

**Review by curdaj — APPROVED** [review]

**Review by dlouhak — APPROVED** [review]

**Pull request merged**